ClientTests/MockProfile.swift

```diff
@@ -72,8 +72,24 @@ open class MockTabQueue: TabQueue {
 }
 
 open class MockProfile: Profile {
+    // Read/Writeable properties for mocking
+    public var recommendations: HistoryRecommendations
+    public var places:  BrowserHistory & Favicons & SyncableHistory & ResettableSyncStorage & HistoryRecommendations
+    public var files: FileAccessor
+    public var history: BrowserHistory & SyncableHistory & ResettableSyncStorage
```

Comment:
In order to assign mock versions of history, recommendations, I needed to move these from getter properties into proper variables which default to the old getter version. The only logical difference here is that they are not lazy loaded.